preprocess/scrub/name_detect.py

```diff
@@ -37,6 +36,29 @@ class CustomNameDetector(scrubadub.detectors.NameDetector):
         "Providence",
     ]
 
-    def __init__(self):
-        for word in self.whitelisted_words:
-            self.disallowed_nouns.add(word)
+    # Spacy's entity labels to be removed
+    name_labels = {"PERSON", "ORG", "GPE"} 
+
+    def __init__(self, client_filth):
+
+        self.client_filth = client_filth
+
+        spacy.prefer_gpu()
+        # load the model
+        self.nlp = spacy.load("en_core_web_lg")
+
+    def scrub(self, text):
+        # run model over description
+        doc = self.nlp(text)
+        for ent in doc.ents:
+            if (ent.label_ in NameDetector.name_labels or ent.text in self.client_filth) and ent.text.lower() not in NameDetector.whitelisted_words:
```

Comment:
Whitelisted words are not being filtered correctly due to inconsistent casing. Maybe we should create a `string_normalize` function for this class that we apply to all the whitelisted words and entities recognized by spacy?